contracts/DolaFlashMinter.sol

```diff
@@ -0,0 +1,98 @@
+//SPDX-License-Identifier: None
+
+pragma solidity ^0.8.0;
+
+import "./interfaces/IERC3156FlashBorrower.sol";
+import "./interfaces/IERC3156FlashLender.sol";
+import "./utils/Ownable.sol";
+import "./utils/Address.sol";
+import "./ERC20/IERC20.sol";
+import "./ERC20/SafeERC20.sol";
+
+contract DolaFlashMinter is Ownable, IERC3156FlashLender {
+    using SafeERC20 for IERC20;
+    event FlashLoan(address receiver, address token, uint256 value);
+    event FlashLoanRateUpdated(uint256 oldRate, uint256 newRate);
+    event TreasuryUpdated(address oldTreasury, address newTreasury);
+
+    IERC20 public constant dola =
+        IERC20(0x865377367054516e17014CcdED1e7d814EDC9ce4);
+    address public treasury = 0x926dF14a23BE491164dCF93f4c468A50ef659D5B;
+    uint256 public flashMinted;
+    uint256 public flashLoanRate = 0.0008 ether;
+
+    bytes32 public immutable CALLBACK_SUCCESS =
+        keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    function flashLoan(
+        IERC3156FlashBorrower receiver,
+        address token,
+        uint256 value,
+        bytes calldata data
+    ) external override returns (bool) {
+        require(token == address(dola), "!dola");
+        require(value <= type(uint112).max, "individual loan limit exceeded");
```

Comment:
Shouldnt we be using maxFlashLoan() here?

Reply:
That's interesting. Wouldn't it be alright to do something like this then:
```solidity
function flashLoan(
        IERC3156FlashBorrower receiver,
        address token,
        uint256 value,
        bytes calldata data
    ) external override returns (bool) {
        require(value <= maxFlashLoan(), "total loan limit exceeded");
        flashMinted = flashMinted + value;

        // ...
}
``` 

Because `maxFlashLoan` already checks for the dola address, and the individual loan limit is equal to the total loan limit?

Reply:
I'd prefer to keep it as is, since it's more clear that each flashloan can up to max uint112, but the total flashloan per block also cannot exceed max uint112.